prover/crates/lib/circuit_prover_service/src/job_runner.rs

```diff
@@ -0,0 +1,142 @@
+use std::{collections::HashMap, sync::Arc};
+
+use shivini::ProverContext;
+use tokio_util::sync::CancellationToken;
+use zksync_object_store::ObjectStore;
+use zksync_prover_dal::{ConnectionPool, Prover};
+use zksync_prover_fri_types::{
+    circuit_definitions::boojum::cs::implementations::setup::FinalizationHintsForProver,
+    get_current_pod_name, ProverServiceDataKey,
+};
+use zksync_prover_job_processor::{Backoff, BackoffAndCancellable, JobRunner};
+use zksync_prover_keystore::GoldilocksGpuProverSetupData;
+use zksync_types::{protocol_version::ProtocolSemanticVersion, prover_dal::FriProverJobMetadata};
+
+use crate::{
+    gpu_circuit_prover::{
+        GpuCircuitProverExecutor, GpuCircuitProverJobPicker, GpuCircuitProverJobSaver,
+    },
+    types::witness_vector_generator_execution_output::WitnessVectorGeneratorExecutionOutput,
+    witness_vector_generator::{
+        HeavyWitnessVectorMetadataLoader, LightWitnessVectorMetadataLoader,
+        WitnessVectorGeneratorExecutor, WitnessVectorGeneratorJobPicker,
+        WitnessVectorGeneratorJobSaver, WitnessVectorMetadataLoader,
+    },
+};
+
+/// Witness Vector Generator runner implementation for light jobs.
+// TODO: Encapsulate arguments, together when refactoring main; same for heavy
```

Comment:
Suggestion: you can define structure `JobBuilder` which would store "common" arguments, and methods for light/heavy wvg and circuit prover runner. So it would be used roughly like this:

```rust
let builder = JobBuilder::new(connection_pool, object_store, finalization_hints_cache, sender, cancellation_token);
let light_wvg = builder.light_wvg(protocol_version, count);
let heavy_wvg = builder.light_wvg(protocol_version, count);
let circuit_prover_runner = builder.circuit_prover_runner(prover_context);
```